app/components/post_list/post/header/expiry_timer/index.tsx

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {Text, View} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import {useTheme} from '@context/theme';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+const TIMER_REFRESH_INTERVAL_MS = 1000;
+
+type Props = {
+    expiryTime: number; // timestamp in ms
+    onExpiry?: () => void;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        display: 'flex',
+        flexDirection: 'row',
+        gap: 6,
+        backgroundColor: changeOpacity(theme.dndIndicator, 0.12),
+        paddingHorizontal: 5,
+        paddingVertical: 2,
+        borderRadius: 4,
+    },
+    text: {
+        fontSize: 12,
+        color: theme.dndIndicator,
+        fontWeight: 600,
+    },
+}));
+
+function formatTime(ms: number) {
```

Comment:
I think there is already a function similar to this somewhere, almost sure is in the video_controls, but perhaps worth unifying.

Reply:
Found it. Moved it to datetime utils and use the same in both places now.